File: cf_96_double_cola.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    int n;
    cin>>n;
    int sum=0,i=0;
    while(n>=sum){
         i+=1;
         sum=5*pow(2,i)-5;
    }
    sum=5*pow(2,(i-1))-5;
    int diff=n-sum;
    int k=pow(2,i-1);
    if(diff>=1 && diff<=(k)){
        cout<<"Sheldon";
    }
    else if (diff>=(k+1) && diff<=2*k){
        cout<<"Leonard";
    }
    else if (diff>=(2*k+1) && diff<=3*k){
        cout<<"Penny";
    }
    else if(diff>=(3*k+1) && diff<=4*k){
        cout<<"Rajesh";
    }
    else{
        cout<<"Howard";
    }
    return 0;
}

File: cf_96_double_cola.py
n=int(input())
sum=0
i=0
while((5*pow(2,i))<=n):
    i+=1
    print(i)
for i in range(i+1):
    sum+=5*(pow(2,i))
    print(sum)
diff=abs(sum-n)
if diff in range(pow(2,i)):
    print("h")
elif diff in range(pow(2,i),pow(2,(i+1))):
    print("r")
elif diff in range(pow(2,(i+1)),pow(2,(i+2))):
    print("p")
elif diff in range(pow(2,(i+2)),pow(2,(i+3))):
    print("l")
else:
    print("s")